ansible/library/fabric_info.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+
+DOCUMENTATION = '''
+module: fabric_info.py
+Ansible_version_added:  2.0.0.2
+short_description:   Find SONiC Fabric ASIC inforamtion if applicable for the DUT
+Description:
+        When the testbed has Fabric ASICs, this module helps to collect that information
+        which helps in generating the minigraph
+    Input:
+        hwsku num_fabric_asic
+
+    Return Ansible_facts:
+    fabric_info:  SONiC Fabric ASIC information
+
+'''
+
+EXAMPLES = '''
+    - name: get Fabric ASIC info
+      fabric_info: hwsku='ACS-MSN2700' num_fabric_asic=1
+'''
+
+RETURN = '''
+      ansible_facts{
+        fabric_info: [{'asicname': 'ASIC0', 'ip_prefix': '10.1.0.33/32', 'ip6_prefix': 'FC00:1::33/128'},
+                      {'asicname': 'ASIC1', 'ip_prefix': '10.1.0.34/32', 'ip6_prefix': 'FC00:1::34/128'}]
+      }
+'''
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            hwsku=dict(required=True, type='str'),
+            num_fabric_asic=dict(type='str', required=False),
+            asics_host_pfx=dict(type='str', required=False),
+            asics_host_pfx6=dict(type='str', required=False)
+        ),
+        supports_check_mode=True
+    )
+    m_args = module.params
+    try:
+        fabric_info = []
+        # num_fabric_asic may not be present for fixed systems which have no Fabric ASIC.
+        # Then return empty fabric_info
+        if 'num_fabric_asic' not in m_args or int(m_args['num_fabric_asic']) < 1:
+           module.exit_json(ansible_facts={'fabric_info': fabric_info})
+           return
+        num_fabric_asic = int( m_args[ 'num_fabric_asic' ] )
+        asics_host_pfx = str( m_args[ 'asics_host_pfx' ] )
+        asics_host_pfx6 = str( m_args[ 'asics_host_pfx6' ] )
+        for asic_id in range(num_fabric_asic):
+            key = "ASIC%d" % asic_id
+            data = { 'asicname': key,
+                     'ip_prefix': asics_host_pfx % asic_id,
+                     'ip6_prefix': asics_host_pfx6 % asic_id }
+            fabric_info.append( data )
+        module.exit_json(ansible_facts={'fabric_info': fabric_info})
+    except (IOError, OSError), e:
+        fail_msg = "IO error" + str(e)
+        module.fail_json(msg=fail_msg)
+    except Exception, e:
+        fail_msg = "failed to find the correct fabric asic info for "+m_args['hwsku'] + str(e)
+        module.fail_json(msg=fail_msg)
+
+from ansible.module_utils.basic import *
+if __name__ == "__main__":
+    main()
```

Comment:
This ansible module is just about building a list of dict based on some facts. It does not really execute anything on remote device. IMO, it is an overkill to make it an ansible module.

Reply:
Today it doesn't execute anything on the dut. But it could do things like get the num_asic from the dut itself when that is supported.